xcp_d/workflow/anatomical.py

```diff
@@ -269,1218 +297,1136 @@ def init_anatomical_wf(
-                        rh_32k_hcpmidthick_resample_wf,
-                        [("out_file", "current_sphere")],
-                    ),
-                    (
-                        rh_native_hcpmidthick_wf,
-                        rh_32k_hcpmidthick_resample_wf,
-                        [("out_file", "in_file")],
-                    ),
-                    (
-                        rh_32k_hcpmidthick_resample_wf,
-                        rh_32k_hcpmidthick_apply_affine,
-                        [("out_file", "in_file")],
-                    ),
-                    (
-                        convert_xfm2world,
-                        rh_32k_hcpmidthick_apply_affine,
-                        [("out_file", "affine")],
-                    ),
-                    (
-                        remerge_warpfield,
-                        rh_32k_hcpmidthick_apply_warpfield,
-                        [("out_file", "forward_warp")],
-                    ),
-                    (
-                        remerge_inv_warpfield,
-                        rh_32k_hcpmidthick_apply_warpfield,
-                        [("out_file", "warpfield")],
-                    ),
-                    (
-                        rh_32k_hcpmidthick_apply_affine,
-                        rh_32k_hcpmidthick_apply_warpfield,
-                        [("out_file", "in_file")],
-                    ),
-                    (
-                        rh_32k_hcpmidthick_apply_warpfield,
-                        ds_hcpmidRsurf_wf,
-                        [("out_file", "in_file")],
-                    ),
-                ]
-            )
+        # fmt:off
+        workflow.connect([
+            (collect_rh_surfaces, ds_rh_standard_space_surfaces, [
+                ("out", "source_file"),
+            ]),
+            (rh_apply_warpfield_to_fsLR32k, ds_rh_standard_space_surfaces, [
+                ("out_file", "in_file"),
+            ]),
+        ])
+        # fmt:on
 
-            workflow.connect(
-                [
-                    (
-                        lh_32k_hcpmidthick_apply_warpfield,
-                        lh_32k_hcpinflated_surf_wf,
-                        [("out_file", "anatomical_surface_in")],
-                    ),
-                    (
-                        lh_32k_hcpinflated_surf_wf,
-                        ds_hcpinfLsurf_wf,
-                        [("inflated_out_file", "in_file")],
-                    ),
-                    (
-                        lh_32k_hcpinflated_surf_wf,
-                        ds_hcpveryinfLsurf_wf,
-                        [("very_inflated_out_file", "in_file")],
-                    ),
-                ]
-            )
+        # make "HCP-style" native midthickness and inflated
+        lh_native_hcpmidthick_wf = pe.Node(
+            SurfaceAverage(),
+            name="lh_native_hcpmidthick_wf",
+            mem_gb=mem_gb,
+            n_procs=omp_nthreads,
+        )

```

Comment:
@madisoth AFAICT, when standard-space surfaces are available (e.g., with the DCAN or HCP pipelines), the HCP-style files will be missing. My assumption is that those pipelines create _just_ the HCP-style files, and they are just mislabeled in XCPD. Is that correct?

If not, should we be creating the HCP-style surface files when standard-space surfaces are already available?

Reply:
Yes, that's correct

Reply:
In that case, it looks like I need to (1) ensure the HCP-style surfaces coming out of the DCAN and HCP conversion functions include `desc-hcp`, (2) check whether datasets have `desc-hcp` or plain standard-space surfaces, and (3) toggle HCP-style surface creation based on that. Thanks for the clarification.

Reply:
I think I need to dig into this a bit more.

1. Is there any situation where the preprocessing workflow might produce fsLR-space midthickness and [very-]inflated files that _aren't_ in the HCP style?
2. If the answer to the above is "yes", would it be better to use the native-space surface files to make HCP-style fsLR-space surfaces (i.e., run the whole workflow anyway), or just output the non-HCP-style fsLR-space surfaces and call it a day?
3. If we have HCP-style midthickness and inflated files, should we also create the very-inflated files? Currently, the workflow doesn't do this, but it seems pretty straightforward.
4. Do we really need to output the non-HCP-style midthickness file? This seems to be the only file that wouldn't match between the two branches of the tree. It gets written out if there are only native-space surfaces, but doesn't if there are standard-space (presumably HCP-style) surfaces.

Reply:
1. Currently no; this would only change if fsLR surface outputs are added to fMRIPrep / NiBabies in the future.
3. DCAN/HCP derivatives should include all 3 (midthickness, inflated, vinflated) in fsLR32k but if inflated/vinflated are missing then it would be good to generate them
4. I'm okay with omitting non-HCP midthickness.

Reply:
>2. DCAN/HCP derivatives should include all 3 (midthickness, inflated, vinflated) in fsLR32k but if inflated/vinflated are missing then it would be good to generate them

The conversion functions don't include vinflated files. Would you mind taking a look at the following and letting me know what the vinflated files would be named?

https://github.com/PennLINC/xcp_d/blob/57fa13f7712f4a552b58202dd83f5b18ca6d7339/xcp_d/utils/hcp2fmriprep.py#L78-L87

Reply:
Resolved in #714.